libmate-desktop/mate-desktop-item.c

```diff
@@ -2003,7 +2008,7 @@ ditem_execute (const MateDesktopItem *item,
 
 	g_free (exec_locale);
 #ifdef HAVE_STARTUP_NOTIFICATION
-	if (sn_context != NULL) {
+	if ((sn_context != NULL) && (GDK_IS_X11_DISPLAY (display))){
```

Comment:
Same here.
https://app.travis-ci.com/github/mate-desktop/mate-desktop/jobs/607773800#L1927
```
mate-desktop-item.c:2011:18: warning: The left operand of '!=' is a garbage value [core.UndefinedBinaryOperatorResult]

        if ((sn_context != NULL) && (GDK_IS_X11_DISPLAY (display))){

             ~~~~~~~~~~ ^
```
in local rpm build.
```
In function 'ditem_execute',
    inlined from 'mate_desktop_item_launch_on_screen_with_env' at mate-desktop-item.c:2131:8:
mate-desktop-item.c:2021:17: warning: 'sn_display' may be used uninitialized [-Wmaybe-uninitialized]
 2021 |                 sn_display_unref (sn_display);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mate-desktop-item.c: In function 'mate_desktop_item_launch_on_screen_with_env':
mate-desktop-item.c:1779:20: note: 'sn_display' was declared here
 1779 |         SnDisplay *sn_display;
      |                    ^~~~~~~~~~
```

Reply:
See
https://github.com/mate-desktop/mate-desktop/pull/587
which initialises `sn_display` to NULL in the wayland case instead of ignoring it.

An alternate approach would be to split the runtime checks , checking x11 first and only if running under x11 running the checks for a valid  `sn_display` . That would save variable being declared (as NULL so not needing to allocate ram for an actual structure) at the price of complicating the source code. The RAM consumption to store the pointer itself (it doesn't point to any address in ram) is said to be architecture-dependent but usually somewhere around 4-8 bytes:
https://stackoverflow.com/questions/32566679/memory-allocated-by-null-pointers

Unless somehow optimized out by the compiler, splitting the checks also has to use more RAM, and almost certainly that would be more than 4-8 bytes so initiializing to NULL probably uses the least ram and the fewest CPU cycles

Reply:
This warning is also gone with https://github.com/mate-desktop/mate-desktop/pull/587